### PR TITLE
add formAdmin struct

### PR DESCRIPTION
**Pull request by MaximeZmt**
Thank you for opening a pull request with this project, please also:

* [ ] add a brief description of your changes here
* [ ] assign the PR to yourself, or to the person(s) working on it
* [ ] start in `draft` mode and `in progress` pipeline in the project (if applicable)
* [ ] if applicable, add this PR to its related issue by one of the special keywords [Closing keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue)
* once it's ready 
  * [ ] put it in the `Review` or `Ready4Merge` pipeline in the project (if applicable) 
  * [ ] remove `draft`
  * [ ] assign a reviewer

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,5 @@ dela/
 bin/
 nodes/
 cookies.txt
+
+.tool-versions
```
Comment (PascalinDe) [inline .gitignore]: what tool generates this? if it is a tool that you use locally, I'd prefer you ignore it locally (https://stackoverflow.com/questions/49305201/gitignore-only-on-local/49306968#49306968)

File: contracts/evoting/admin_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+func init() {
+	ciphervoteFac := types.CiphervoteFactory{}
+	formFacAdminTest = types.NewFormFactory(ciphervoteFac, fakeAuthorityFactory{})
+	transactionFacAdminTest = types.NewTransactionFactory(ciphervoteFac)
+
+	ctxAdminTest = sjson.NewContext()
+}
+
+func TestAdmin_serde(t *testing.T) {
```
Comment (PascalinDe) [inline contracts/evoting/admin_test.go:27]: I think it's helpful to add a comment to explain what you're testing, that makes it easier to get into the context when debugging failing tests

File: contracts/evoting/json/adminForm.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			FormID:    m.FormID,
+			AdminList: m.AdminList,
+		}
+
+		buff, err := ctx.Marshal(&adminFormJSON)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to marshal form: %v", err)
+		}
+
+		return buff, nil
```
Comment (PascalinDe) [inline contracts/evoting/json/adminForm.go]: I take it the idea behind the `switch` statement is to add other formats at a later point?

File: web/backend/src/controllers/dela.ts [hunk 4]
```diff
@@ -265,6 +265,5 @@ delaRouter.use('/*', (req, res) => {
   }
 
   const dataStr = JSON.stringify(bodyData);
-
```
Comment (PascalinDe) [inline web/backend/src/controllers/dela.ts]: is this change on purpose? if not please remove it from the PR :)

Comment (MaximeZmt) [reply web/backend/src/controllers/dela.ts]: Indeed I will revert it 👍

File: contracts/evoting/json/adminForm.go (hunk 3 above)
Comment (MaximeZmt) [reply contracts/evoting/json/adminForm.go]: I just took the same coding style/convention of the equivalent file for the "classic" form: https://github.com/MaximeZmt/d-voting/blob/main/contracts/evoting/json/forms.go

Should I change it ?

File: contracts/evoting/admin_test.go (hunk 2 above)
Comment (MaximeZmt) [reply contracts/evoting/admin_test.go:27]: Indeed will add it

File: .gitignore (hunk 1 above)
Comment (MaximeZmt) [reply .gitignore]: Most likely my IDE, noted, will exclude it locally 👍

File: contracts/evoting/json/adminForm.go (hunk 3 above)
Comment (PascalinDe) [reply contracts/evoting/json/adminForm.go]: if there's only one format the `switch` logic seems to be a bit superfluous to me, especially as you're not doing it in the `Decode` function - I'd just do something like this:

```
func (adminFormFormat) Encode(ctx serde.Context, message serde.Message) ([]byte, error) {
    if message.(type) != types.AdminForm {
        return nil, xerrors.Errorf("Unknown format: %T", message)
    }
    <rest of the function>
}
```
like this it returns immediately if it's not the correct type (no need for an `else` which makes it more readable), and the logic is simpler (makes it easier to understand)

`switch` makes only sense if there's multiple cases

File: contracts/evoting/json/adminForm.go [hunk 5]
```diff
@@ -9,22 +9,22 @@ import (
...
-
-		buff, err := ctx.Marshal(&adminFormJSON)
-		if err != nil {
-			return nil, xerrors.Errorf("failed to marshal form: %v", err)
-		}
-
-		return buff, nil
-	default:
+	adminForm, ok := message.(types.AdminForm)
+	if !ok {
```
Comment (PascalinDe) [inline contracts/evoting/json/adminForm.go:13]: if there is no error it is the correct format?

**Review by PascalinDe — APPROVED** [review]
LGTM
